src/lib/ClaimHashLib.sol

```diff
@@ -61,91 +61,54 @@ library ClaimHashLib {
         view
         returns (bytes32 claimHash, bytes32 typehash)
     {
-        return _toMultichainClaimHashAndTypehash(claim);
+        uint256 commitmentsHash = HashLib.toCommitmentsHashFromSingleLock(claim.asRawPtr());
+        (bytes32 allocationTypehash, bytes32 compactTypehash) = claim.asRawPtr().toMultichainTypehashes();
+        return (
+            HashLib.toMultichainClaimHash(claim.asRawPtr(), 0xa0, allocationTypehash, compactTypehash, commitmentsHash),
```

Comment:
This too (especially if we're not doing function casts anymore which is why they were needed previously IIRC)